flink-ml-lib/src/main/java/org/apache/flink/ml/feature/onehotencoder/OneHotEncoderModel.java

```diff
@@ -0,0 +1,217 @@
+        return new Table[] {outputTable};
+    }
+
+    @Override
+    public void save(String path) throws IOException {
+        StreamTableEnvironment tEnv =
+                (StreamTableEnvironment) ((TableImpl) modelTable).getTableEnvironment();
+
+        String dataPath = ReadWriteUtils.getDataPath(path);
+        FileSink<Tuple2<Integer, Integer>> sink =
+                FileSink.forRowFormat(
+                                new Path(dataPath), new OneHotEncoderModelData.ModelDataEncoder())
+                        .withRollingPolicy(OnCheckpointRollingPolicy.build())
+                        .withBucketAssigner(new BasePathBucketAssigner<>())
+                        .build();
+        OneHotEncoderModelData.getModelDataStream(tEnv, modelTable).sinkTo(sink);
+
+        ReadWriteUtils.saveMetadata(this, path);
+    }
+
+    public static OneHotEncoderModel load(StreamExecutionEnvironment env, String path)
+            throws IOException {
+        StreamTableEnvironment tEnv = StreamTableEnvironment.create(env);
+
+        Source<Tuple2<Integer, Integer>, ?, ?> source =
+                FileSource.forRecordStreamFormat(
+                                new OneHotEncoderModelData.ModelDataStreamFormat(),
+                                ReadWriteUtils.getDataPaths(path))
+                        .build();
+        OneHotEncoderModel model = ReadWriteUtils.loadStageParam(path);
+        DataStream<Tuple2<Integer, Integer>> modelData =
+                env.fromSource(source, WatermarkStrategy.noWatermarks(), "modelData");
+        model.setModelData(OneHotEncoderModelData.getModelDataTable(tEnv, modelData));
+
+        return model;
+    }
+
+    @Override
+    public Map<Param<?>, Object> getParamMap() {
+        return paramMap;
+    }
+
+    @Override
+    public OneHotEncoderModel setModelData(Table... inputs) {
+        Preconditions.checkArgument(inputs.length == 1);
+        modelTable = inputs[0];
+        return this;
+    }
+
+    @Override
+    public Table[] getModelData() {
+        return new Table[] {modelTable};
+    }
+
+    private static class PredictLabelOperator
+            extends AbstractUdfStreamOperator<Row, AbstractRichFunction>
+            implements OneInputStreamOperator<Row, Row> {
+        private final String[] inputCols;
+        private final boolean dropLast;
+        private final String broadcastModelKey;
+
+        public PredictLabelOperator(
+                String[] inputCols, boolean dropLast, String broadcastModelKey) {
+            super(new AbstractRichFunction() {});
+            this.inputCols = inputCols;
+            this.dropLast = dropLast;
+            this.broadcastModelKey = broadcastModelKey;
+        }
+
+        @Override
+        public void processElement(StreamRecord<Row> streamRecord) {
+            Row row = streamRecord.getValue();
+            List<Tuple2<Integer, Integer>> model =
+                    userFunction.getRuntimeContext().getBroadcastVariable(broadcastModelKey);
```

Comment:
Can we call `getRuntimeContext().getBroadcastVariable(broadcastModelKey)` directly?

Could this class extends `AbstractStreamOperator` instead of `AbstractUdfStreamOperator` for simplicity?

Reply:
According to the offline discussion with Zhipeng, we can directly use `RichMapFunction` here. I'll make the change in this way.